kafka/integration_test.go

```diff
@@ -828,21 +828,135 @@ func (its *IntegrationTestSuite) TestAdminClient_DeleteConsumerGroups() {
 		t.Errorf("Consumer group %s should not be present\n", groupID)
 		return
 	}
+	return
 }
 
 // TestAdminClient_ListAndDescribeConsumerGroups validates the working of the
 // list consumer groups and describe consumer group APIs of the admin client.
 //
-// We test the following situations:
+// We test the following situations[Classic Protocol]:
 //
 // 1. One consumer group with one client.
 // 2. One consumer group with two clients.
 // 3. Empty consumer group.
+//
+// We test the following situations[Consumer Protocol]:
+// 1. No Consumer Group should be listed with Classic Group Type Option
+// 2. Our Consumer Group should be listed with Consumer Group Type Option
+// 3. All Consumer Groups should be listed with NULL Group Type Option
 func (its *IntegrationTestSuite) TestAdminClient_ListAndDescribeConsumerGroups() {
 	t := its.T()
 	if !testConsumerGroupProtocolClassic() {
-		t.Skipf("KIP 848 Admin operations changes still aren't " +
-			"available")
+		// Generating a new topic/groupID to ensure a fresh group/topic is created.
+		rand.Seed(time.Now().Unix())
+		groupID := fmt.Sprintf("%s-%d", testconf.GroupID, rand.Int())
+		topic := fmt.Sprintf("%s-%d", testconf.TopicName, rand.Int())
+		nonExistentGroupID := fmt.Sprintf("%s-nonexistent-%d", testconf.GroupID, rand.Int())
```

Comment:
Move above the "if" block, and remove from below too, so they're not duplicated

Reply:
Same for creation of adminclient, topic creation, etc.

Reply:
Actually this is done voluntarily because currently in semaphore we run the whole tests with broker version < 3.8.0.0 and then when we run with higher broker version on semaphore to test the consumer group protocol we only check the case of consumer group that is why we return at the end.

Reply:
Okay, so since there is nothing common between the tests, why don't you split them? I don't see why we should have an entirely different test in an if block if there's any shared code between them, it just makes it harder to read.

TestAdminClient_ListAndDescribeConsumerGroupsClassic and TestAdminClient_ListConsumerGroupsConsumer